### PR TITLE
[BZ2004651] Additional day 2 steps for external load balancers with Octavia  

### PR DESCRIPTION
**Pull request by maxwelldb**
Context: https://bugzilla.redhat.com/show_bug.cgi?id=2004651

Preview: https://deploy-preview-40581--osdocs.netlify.app/openshift-enterprise/latest/post_installation_configuration/network-configuration.html#nw-osp-configure-octavia-load-balancing-services_post-install-network-configuration

### CODE REVIEW

**Comment by maxwelldb** [comment]
@iamemilio Is this about what you were thinking? 

https://deploy-preview-40581--osdocs.netlify.app/openshift-enterprise/latest/post_installation_configuration/network-configuration.html#nw-osp-configure-octavia-load-balancing-services_post-install-network-configuration

**Comment by maxwelldb** [comment]
@pierreprinetti mentioning you here, since you own https://bugzilla.redhat.com/show_bug.cgi?id=2004236 at the moment.

File: modules/nw-osp-configure-octavia-load-balancing-services.adoc [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+You can create load balancer service types and ingress controllers that have load balancers as backends by using Octavia on {rh-openstack-first}. 
+
+[NOTE]
+====
+Services and controllers that rely on Octavia have the following limitations:
+
+* Only TCP traffic is supported.
+* Active Octavia load balancers and the floating IP addresses that are attached to them are not deleted during a cluster delete operation. They must be deleted beforehand.
+* The `manage-security-groups` property in the cloud provider configuration doesn't not apply to non-administrator tenants.
```
Comment (pierreprinetti) [inline modules/nw-osp-configure-octavia-load-balancing-services.adoc:16]: `doesn't not` looks like a looks

**Comment by pierreprinetti** [comment]
Except for the nit, LGTM. @mandre can you have a look as well please?

**Comment by mandre** [comment]
LGTM. This is going only into the 4.8 docs, correct?

File: modules/nw-osp-configure-octavia-load-balancing-services.adoc (hunk 1 above)
Comment (maxwelldb) [reply modules/nw-osp-configure-octavia-load-balancing-services.adoc:16]: Well spotted. 

```suggestion
* The `manage-security-groups` property in the cloud provider configuration does not apply to non-administrator tenants.
```

**Comment by maxwelldb** [comment]
FYI @eurijon -- this is the downstream version of https://bugzilla.redhat.com/show_bug.cgi?id=2004236

**Comment by pierreprinetti** [comment]
> @mandre If you and/or @pierreprinetti say so, yeah. :)
> 
> 4.8 only -> target `enterprise-4.8` 4.8 and later -> target `main`, cherry-pick to 4.8+

We added an automated process in 4.9; these instructions only apply to 4.8 (again, @mandre please confirm!)

File: modules/nw-osp-configure-octavia-load-balancing-services.adoc (hunk 1 above)
Comment (pierreprinetti) [reply modules/nw-osp-configure-octavia-load-balancing-services.adoc:16]: "administrator tenant" is confusing. I would opt for something like this instead:

```
The `manage-security-groups` property in the cloud provider configuration only applies to OpenStack tenants with administrative privileges.
```

**Comment by maxwelldb** [comment]
TODO: Change merge target.

File: modules/nw-osp-configure-octavia-load-balancing-services.adoc (hunk 1 above)
Comment (maxwelldb) [reply modules/nw-osp-configure-octavia-load-balancing-services.adoc:16]:
```suggestion
* The `manage-security-groups` property in the cloud provider configuration only applies to {rh-openstack} tenants that have administrative privileges. 
```

TY!

**Comment by mandre** [comment]
Y

> > @mandre If you and/or @pierreprinetti say so, yeah. :)
> > 4.8 only -> target `enterprise-4.8` 4.8 and later -> target `main`, cherry-pick to 4.8+
> 
> We added an automated process in 4.9; these instructions only apply to 4.8 (again, @mandre please confirm!)

Yes, sorry for the lag. I confirm it's for 4.8 only.

**Comment by eurijon** [comment]
@maxwelldb hey, it looks good from QE perspective, I guess we'll need it in 4.6 and 4.7 too
cc: @mandre @pierreprinetti @dulek

**Comment by maxwelldb** [comment]
Moved BZ to Verified. 

Waiting on version confirmation from devs.

File: modules/nw-osp-configure-octavia-load-balancing-services.adoc [hunk 2]
```diff
@@ -0,0 +1,72 @@
+// Module included in the following assemblies:
+//
+// * some post-install one...
```
Comment (maxwelldb) [inline modules/nw-osp-configure-octavia-load-balancing-services.adoc:3]:
```suggestion
// * post_installation_configuration/network-configuration.adoc
```

**Comment by maxwelldb** [comment]
@mandre @pierreprinetti @dulek Any thoughts on [this version comment](https://github.com/openshift/openshift-docs/pull/40581#issuecomment-1048835942)?

**Comment by mandre** [comment]
> @mandre @pierreprinetti @dulek Any thoughts on [this version comment](https://github.com/openshift/openshift-docs/pull/40581#issuecomment-1048835942)?

Will defer to @eurijon, depending on the coverage of LB testing in 4.6 and 4.7.

**Comment by eurijon** [comment]
We are testing LB type services and LB type IngressControllers in 4.6, 4.7, 4.8, 4.9 and 4.10 now, for all the network types, and although it's supported there are some BZs:

- https://bugzilla.redhat.com/show_bug.cgi?id=2018481
- https://bugzilla.redhat.com/show_bug.cgi?id=2057526
- https://bugzilla.redhat.com/show_bug.cgi?id=2056605

**Comment by maxwelldb** [comment]
My preference would be not to extend this back to versions where testing or stability are iffy. What do you think, @mandre?

File: modules/nw-osp-configure-octavia-load-balancing-services.adoc [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+use-octavia = true
+lb-provider = ovn
+lb-method = SOURCE_IP_PORT
+----
++
+[NOTE]
+====
+If you are using the OVN driver for Octavia, you must also modify the TCP ingress security group rules for the master and worker security groups to allow IPv4 traffic to ports 30000 through 32767 from 0.0.0.0/0.
+====
+
```
Comment (maxwelldb) [inline modules/nw-osp-configure-octavia-load-balancing-services.adoc:60]:
```suggestion
```

File: modules/nw-osp-configure-octavia-load-balancing-services.adoc [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+----
+[LoadBalancer]
+use-octavia = true
+lb-provider = ovn
+lb-method = SOURCE_IP_PORT
+----
++
+[NOTE]
+====
+If you are using the OVN driver for Octavia, you must also modify the TCP ingress security group rules for the master and worker security groups to allow IPv4 traffic to ports 30000 through 32767 from 0.0.0.0/0.
```
Comment (EricPonvelle) [inline modules/nw-osp-configure-octavia-load-balancing-services.adoc:58]: s/master/primary

Per the ISG

File: modules/nw-osp-configure-octavia-load-balancing-services.adoc [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+====
+
+. If you have multiple external networks, set the value of the `floating-network-id` parameter in your cloud provider configuration to the UUID of the external network in which floating IP addresses are created. For example:
```
Comment (EricPonvelle) [inline modules/nw-osp-configure-octavia-load-balancing-services.adoc:61]: "are created" is passive, but I think this is an exception mentioned in the style guide.

**Review by EricPonvelle — COMMENTED** [review]
I had some minor items that needed attention.

File: modules/nw-osp-configure-octavia-load-balancing-services.adoc [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+= Enabling {rh-openstack} Octavia for load balancer services
+
+You can create load balancer service types and ingress controllers that have load balancers as back ends by using Octavia on {rh-openstack-first}. 
+
+[NOTE]
+====
+Services and controllers that rely on Octavia have the following limitations:
+
+* Only TCP traffic is supported.
+* Active Octavia load balancers and the floating IP addresses that are attached to them are not deleted during a cluster delete operation. They must be deleted beforehand.
```
[the remark this answers is not in the document]
Comment (maxwelldb) [reply modules/nw-osp-configure-octavia-load-balancing-services.adoc:15]:
```suggestion
* Active Octavia load balancers and the floating IP addresses that are attached to them are not deleted during a cluster delete operation. You must delete these items prior performing the operation.
```

File: modules/nw-osp-configure-octavia-load-balancing-services.adoc (hunk 4 above)
Comment (maxwelldb) [reply modules/nw-osp-configure-octavia-load-balancing-services.adoc:58]:
```suggestion
If you are using the OVN driver for Octavia, you must also modify the TCP ingress security group rules for the primary and worker security groups to allow IPv4 traffic to ports 30000 through 32767 from 0.0.0.0/0.
```

**Pull request merged**